src/tests/Regressions/coreclr/GitHub_116953/test116953.cs

```diff
@@ -38,7 +38,7 @@ public class Test116953
     public static Assembly AssemblyA;
     public static Assembly AssemblyC;
 
-    [Fact]
+    [ConditionalFact(typeof(TestLibrary.Utilities), nameof(TestLibrary.Utilities.IsNotNativeAot))]
```

Comment:
```suggestion
    [ConditionalFact(typeof(TestLibrary.Utilities), nameof(TestLibrary.Utilities.HasAssemblyFiles))]
```
?

Reply:
I agree with you that `HasAssemblyFiles` is an incomplete condition, and it really needs to be `HasAssemblyFiles` and `DynamicLoadingSupported` and maybe even more things.

Where possible, I think it is better to use incomplete feature conditions to make things work and refine them as necessary vs. disabling tests on specific runtime flavors.